src/TextField/TextField.js

```diff
@@ -0,0 +1,266 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { dcnb } from "cnbuilder";
+import { MailIcon, XIcon, CheckIcon } from "@heroicons/react/outline";
+import { labelWeights, iconAlignment } from "./TextField.levers";
+
+/**
+ * Text Input  Component
+ *
+ */
+export const TextField = (
+  {
+    className,
+    id,
+    label,
+    fontWeight,
+    helperText,
+    placeholder,
+    icon,
+    type,
+    maxLength,
+    minLength,
+    errorText,
+    validText,
+    isError,
+    isValid,
+    isIcon,
+    iconPosition,
+    isDisabled,
+    isRequired,
+    showCharCount,
+  },
+  props
```

Comment:
Is this correct to have additional props coming in as a second argument the component function? I've not come across this pattern before.